torchao/experimental/ops/tests/test_linear_8bit_act_xbit_weight.cpp

```diff
@@ -36,14 +41,40 @@ UKernelConfig get_ukernel_config() {
       &ukernel::prepare_weight_data<weight_nbit, has_weight_zeros, has_bias>;
   config.kernel_fn =
       &ukernel::kernel<weight_nbit, has_weight_zeros, has_bias, has_clamp>;
+  } else {
+#if defined(TORCHAO_ENABLE_KLEIDI)
+    assert (weight_nbit == 4);
```

Comment:
This is fine in test, but asserts get removed in release build, so use TORCHAO_CHECK

Reply:
we should run tests in debug mode, but for the proper integration (which is not done in this PR) we should check and allow taking kleidi path only when `if constexpr (weight_4bit == 4)`.